### ISSUE TITLE
feat: native Tensor<Complex<T>> support + GPU kernels for HRE spectral acceleration

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet.Tensors#127]
## Summary

The Harmonic Resonance Engine (ooples/AiDotNet#1108) requires native \`Tensor<Complex<T>>\` support throughout the IEngine stack — CPU SIMD and GPU kernels — to achieve production performance. The current interleaved \`Tensor<T>\` format for complex numbers is a performance bottleneck because:

1. **SIMD can't vectorize interleaved data** — real and imaginary parts alternate, preventing packed float operations
2. **Conversion overhead** — every FFT call requires converting between \`Complex<T>\` and interleaved format
3. **GPU memory layout** — interleaved prevents coalesced memory access patterns

### Core Type Support

\`Tensor<Complex<T>>\` should be a first-class tensor type with:
- Contiguous storage of \`Complex<T>\` elements (each element is a struct with \`.Real\` and \`.Imaginary\`)
- Shape semantics identical to \`Tensor<T>\` — a \`Tensor<Complex<T>>\` of shape \`[N]\` has N complex elements
- Automatic SIMD vectorization where the runtime supports \`Vector<Complex<T>>\`

#### FFT Operations (GPU-accelerable via existing CudaFFTKernels)

| Operation | Signature | Notes |
|-----------|-----------|-------|
| \`FFTForward\` | \`Tensor<Complex<T>> FFTForward<T>(Tensor<T> input)\` | Real → Complex FFT |
| \`FFTInverse\` | \`Tensor<T> FFTInverse<T>(Tensor<Complex<T>> input)\` | Complex → Real IFFT |
| \`FFTForwardComplex\` | \`Tensor<Complex<T>> FFTForwardComplex<T>(Tensor<Complex<T>> input)\` | Complex → Complex FFT |
| \`FFTInverseComplex\` | \`Tensor<Complex<T>> FFTInverseComplex<T>(Tensor<Complex<T>> input)\` | Complex → Complex IFFT |

GPU path: Wire to existing \`CudaFFTKernels.cs\` (bit_reverse_permutation + fft_butterfly kernels already exist).

#### Complex Arithmetic (element-wise, GPU-parallelizable)

| Operation | Signature | GPU Kernel Needed |
|-----------|-----------|-------------------|
| \`ComplexMultiply\` | \`Tensor<Complex<T>> x Tensor<Complex<T>> → Tensor<Complex<T>>\` | Yes: \`(a.re*b.re - a.im*b.im, a.re*b.im + a.im*b.re)\` |
| \`ComplexConjugate\` | \`Tensor<Complex<T>> → Tensor<Complex<T>>\` | Yes: \`(re, -im)\` per element |
| \`ComplexMagnitude\` | \`Tensor<Complex<T>> → Tensor<T>\` | Yes: \`sqrt(re² + im²)\` per element |
| \`ComplexPhase\` | \`Tensor<Complex<T>> → Tensor<T>\` | Yes: \`atan2(im, re)\` per element |
| \`ComplexFromPolar\` | \`(Tensor<T> mag, Tensor<T> phase) → Tensor<Complex<T>>\` | Yes: \`(mag*cos(phase), mag*sin(phase))\` |
| \`ComplexMagnitudeSquared\` | \`Tensor<Complex<T>> → Tensor<T>\` | Yes: \`re² + im²\` (avoids sqrt) |
| \`ComplexScale\` | \`(Tensor<Complex<T>>, T scalar) → Tensor<Complex<T>>\` | Yes: \`(re*s, im*s)\` |
| \`ComplexAdd\` | \`Tensor<Complex<T>> + Tensor<Complex<T>> → Tensor<Complex<T>>\` | Yes: element-wise add |

#### Support Operations

| Operation | Signature | Notes |
|-----------|-----------|-------|
| \`TensorTopKComplex\` | \`(Tensor<Complex<T>>, int k) → Tensor<Complex<T>>\` | Top-K by magnitude, zero rest |
| \`TensorSoftmaxRows\` | \`Tensor<T> [M,N] → Tensor<T> [M,N]\` | Per-row softmax (for attention) |

### GPU Kernel Requirements

Each backend (CUDA, OpenCL, Metal, Vulkan, HIP, WebGPU) needs kernels for:

1. **Complex element-wise ops** — trivially parallelizable (1 thread per complex element)
2. **FFT butterflies** — already implemented in \`CudaFFTKernels.cs\`, need to wire to \`Tensor<Complex<T>>\`
3. **Complex magnitude/phase** — simple per-element kernels

CUDA kernel template (for reference):
\`\`\`cuda
extern "C" __global__ void complex_multiply(
    const float2* a, const float2* b, float2* result, int n)
{
    int idx = blockIdx.x * blockDim.x + threadIdx.x;
    if (idx >= n) return;
    float2 va = a[idx], vb = b[idx];
    result[idx].x = va.x * vb.x - va.y * vb.y;  // re
    result[idx].y = va.x * vb.y + va.y * vb.x;  // im
}
\`\`\`

### CpuEngine Implementation

- Iterative Cooley-Tukey FFT (already exists in CpuEngine for interleaved format, adapt for \`Complex<T>\`)
- SIMD-accelerated complex arithmetic where possible
- Fallback scalar loops for \`Complex<T>\` on platforms without SIMD support

## Priority

1. **Tensor<Complex<T>> type support** — foundation for everything else
2. **FFT operations** — most critical path (called on every HRE forward pass)
3. **Complex element-wise ops** — enables vectorized spectral filtering
4. **GPU kernels** — enables GPU-accelerated HRE inference

## Context

The Harmonic Resonance Engine replaces traditional neural network weights with spectral communication via FFT. Every forward pass involves multiple FFTs, complex multiplications, and magnitude extractions. Without native \`Tensor<Complex<T>>\` support, all spectral operations fall back to scalar \`NumOps\` loops — negating the O(N log N) complexity advantage.

See ooples/AiDotNet#1108 for the HRE implementation.
